app/models/spree/order_contents_decorator.rb

```diff
@@ -1,5 +1,26 @@
 module Spree
   OrderContents.class_eval do
+
+    def add_to_line_item(variant, quantity, options = {})
+      line_item = grab_line_item_by_variant(variant, false, options)
+
+      if line_item && part_variants_match(line_item, variant, quantity, options)
+        line_item.quantity += quantity.to_i
+        line_item.currency = currency unless currency.nil?
```

Comment:
That's pulled straight from spree haha. I just added && part_variants_match

On Tue, Aug 25, 2015 at 6:29 AM Prem Sichanugrist <notifications@github.com>
wrote:

> In app/models/spree/order_contents_decorator.rb
> <https://github.com/spree-contrib/spree-product-assembly/pull/121#discussion_r37864231>
> :
>
> > @@ -1,5 +1,26 @@
> >  module Spree
> >    OrderContents.class_eval do
> > +
> > +    def add_to_line_item(variant, quantity, options = {})
> > +      line_item = grab_line_item_by_variant(variant, false, options)
> > +
> > +      if line_item && part_variants_match(line_item, variant, quantity, options)
> > +        line_item.quantity += quantity.to_i
> > +        line_item.currency = currency unless currency.nil?
>
> YAGNI on the conditional. You'll assign a nil regardless?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spree-contrib/spree-product-assembly/pull/121/files#r37864231>
> .
>